core/lib/zksync_core/src/state_keeper/keeper.rs

```diff
@@ -136,25 +139,50 @@ impl ZkSyncStateKeeper {
 
         let previous_batch_protocol_version =
             self.io.load_previous_batch_version_id().await.unwrap();
+
+        let first_batch_in_shared_bridge = l1_batch_env.number == L1BatchNumber(1)
+            && protocol_version >= ProtocolVersionId::Version21;
```

Comment:
Let's add a method for `ProtocolVersionId` called `is_post_shared_bridge` or something and let it return false for now as we don't know if we would need to do another upgrade before shared bridge.